driver/reducecalculation_driver.hpp

```diff
@@ -88,18 +83,56 @@ int32_t mloReduceCalculationForwardRunHost(miopenTensorDescriptor_t inputDesc,
     }
     return ret;
 }
-#endif
 
-template <typename Tgpu, typename Tref>
+template <typename Tgpu, typename Tcheck, ReduceCalculationOp_t op>
+int32_t mloReduceLogicalCalculationForwardRunHost(miopenTensorDescriptor_t inputDesc,
+                                                  miopenTensorDescriptor_t outputDesc,
+                                                  Tgpu* input,
+                                                  Tcheck* outputhost,
+                                                  int32_t dim)
+{
+    auto input_dims  = miopen::deref(inputDesc).GetLengths();
+    auto output_dims = miopen::deref(outputDesc).GetLengths();
+
+    auto reduce_size = input_dims[dim];
+    auto output_numel =
+        std::accumulate(output_dims.begin(), output_dims.end(), 1LL, std::multiplies<int64_t>());
+
+    auto inner_size = 1ULL;
+    for(int32_t i = dim + 1; i < input_dims.size(); ++i)
```

Comment:
this can lead to CICD failed when i is int32_t while input_dims.size() is size_t

Reply:
fixed: change type of `i` to `size_t`